FE/angular/src/app/components/movie-card/movie-card.component.spec.ts

```diff
@@ -2,22 +2,64 @@ import { ComponentFixture, TestBed } from '@angular/core/testing';
 
 import { MovieCardComponent } from './movie-card.component';
 import { GoDetailsComponent } from '../navigation/go-details/go-details.component';
+import { Spectator, createComponentFactory } from '@ngneat/spectator';
 
 describe('MovieCardComponent', () => {
+  let spectator: Spectator<MovieCardComponent>;
   let component: MovieCardComponent;
-  let fixture: ComponentFixture<MovieCardComponent>;
+  const createComponent = createComponentFactory({
+    component: MovieCardComponent,
+    imports: [],
+    declarations: [],
```

Comment:
You are missing declarations for the navigation elements. So `<app-go-imdb>` and `<app-go-details>` will not render.
Your tests do not catch these.